stdlib/public/TensorFlow/Parameterized.swift

```diff
@@ -65,3 +65,59 @@ public extension Parameterized where Parameters : ParameterGroup {
     allParameters.update(withGradients: gradients, updater)
   }
 }
+
+//===----------------------------------------------------------------------===//
+// Extend common types to conform to ParameterGroup and Parameterized
+//===----------------------------------------------------------------------===//
+
+/// Tensors conform to the protocols by "identity".
+extension Tensor : Parameterized {
+  public typealias Parameters = Tensor<Scalar>
+
+  public var allParameters: Parameters {
+    get {
+      return self
+    }
+    set(newValue) {
+      self = newValue
+    }
+  }
+}
+
+extension Tensor : ParameterGroup {
+  public typealias Parameter = Tensor<Scalar>
+
+  public mutating func update(
+      withGradients gradients: Tensor<Scalar>,
+      _ updateParameter: (inout Parameter, Parameter) -> Void) {
+    updateParameter(&self, gradients)
+  }
+}
+
+/// Arrays conform to the protocols if their elements conform to them.
+extension Array : Parameterized where Element: Parameterized {
+  public typealias Parameters = [Element.Parameters]
+
+  public var allParameters: Parameters {
+    get {
+      return self.map{ $0.allParameters }
+    }
+    set(newParameters) {
+      for idx in self.indices {
+        self[idx].allParameters = newParameters[idx]
+      }
+    }
+  }
+}
+
+extension Array : ParameterGroup where Element: ParameterGroup  {
```

Comment:
The standard library's coding convention expects a space before `:` in generic constraints.

```suggestion
extension Array : ParameterGroup where Element : ParameterGroup  {
```